cpp/tests/reductions/scan_tests.cu

```diff
@@ -0,0 +1,193 @@
+
+// This is the main test feature
+template <typename T>
+struct ScanTest : public cudf::test::BaseFixture
+{
+    void scan_test(
+        cudf::test::fixed_width_column_wrapper<T> const col_in,
+        cudf::test::fixed_width_column_wrapper<T> const expected_col_out,
+        scan_op op, bool inclusive)
+    {
+        bool do_print = false;
+
+        auto int_values = cudf::test::to_host<T>(col_in);
+        auto exact_values = cudf::test::to_host<T>(expected_col_out);
+        this->val_check(std::get<0>(int_values), do_print, "input = ");
+        this->val_check(std::get<0>(exact_values), do_print, "exact = ");
+
+        const column_view input_view = col_in;
+        std::unique_ptr<cudf::column> col_out;
+
+        CUDF_EXPECT_NO_THROW( col_out = cudf::experimental::scan(input_view, op, inclusive) );
+        const column_view result_view = col_out->view();
+
+        cudf::test::expect_column_properties_equal(input_view, result_view);
+        cudf::test::expect_columns_equal(expected_col_out, result_view);
+
+        auto host_result = cudf::test::to_host<T>(result_view);
+        this->val_check(std::get<0>(host_result), do_print, "result = ");
+    }
+
+    template <typename Ti>
+    void val_check(std::vector<Ti> const & v, bool do_print=false, const char* msg = nullptr){
+        if( do_print ){
+            std::cout << msg << " {";
+            std::for_each(v.begin(), v.end(), [](Ti i){ std::cout << ", " <<  i;});
+            std::cout << "}"  << std::endl;
+        }
+        range_check(v);
+    }
+
+    // make sure all elements in the range of sint8([-128, 127])
+    template <typename Ti>
+    void range_check(std::vector<Ti> const & v){
+        std::for_each(v.begin(), v.end(),
+            [](Ti i){
+                ASSERT_GE(static_cast<int>(i), -128);
+                ASSERT_LT(static_cast<int>(i),  128);
+            });
+    }
+
+};
+
+using Types = cudf::test::NumericTypes;
+//using Types = testing::Types<int32_t>;
+
+TYPED_TEST_CASE(ScanTest, Types);
+
+// ------------------------------------------------------------------------
+TYPED_TEST(ScanTest, Min)
+{
+    std::vector<TypeParam>  v({123, 64, 63, 99, -5, 123, -16, -120, -111});
+    std::vector<bool> b({  1,  0,  1,  1,  1,   1,   0,    1,    1});
+    std::vector<TypeParam> exact(v.size());
+
+    std::transform(v.cbegin(), v.cend(),
+        exact.begin(),
+        [acc=v[0]](auto i) mutable { acc = std::min(acc, i); return acc; }
+        );
+
+    this->scan_test({v.begin(), v.end()}, 
+                    {exact.begin(), exact.end()},
+                    scan_op::MIN, true);
+
+    std::transform(v.cbegin(), v.cend(), b.begin(),
+        exact.begin(),
+        [acc=v[0]](auto i, bool b) mutable { if(b) acc = std::min(acc, i); return acc; }
+        );
+
+    this->scan_test({v.begin(), v.end(), b.begin()}, 
+                    {exact.begin(), exact.end(), b.begin()},
+                    scan_op::MIN, true);
+}
+
+TYPED_TEST(ScanTest, Max)
```

Comment:
What about Min/Max scan with strings? I guess the support is not in yet since it requires a two-pass approach?

Reply:
Yes. Is there any post processing functions similar to our legacy string column?

Reply:
Not sure what you mean. You mean like NVCategory_gather?  No. I think you would need to do a scan that outputs the size at each location and then use that to allocate the string column. Then scan again, storing the min/max string in each element as you go.  I have the same exact problem in rolling window.  It can be a later PR.

Reply:
Please open an issue though.

Reply:
https://github.com/rapidsai/cudf/issues/3513